odl/space/cu_ntuples.py

```diff
@@ -847,36 +976,271 @@ def _norm_default(x):
     return x.data.norm()
 
 
+def _pnorm_default(x, p):
+    if p == float('inf'):
+        raise NotImplementedError('inf-norm not implemented.')
+    elif int(p) != p:
+        raise NotImplementedError('non-integer powers not implemented')
+    # TODO: optimized version in C++ code?
```

Comment:
You can use a `thrust::transform_reduce`, but since you will have three input vectors (at least in the `inner` case), you will need to wrap the data using `thrust::zip_iterator` and use a custom functor I think.

We could do this come monday, ok?